app/charts/shared/imputation.tsx

```diff
@@ -0,0 +1,82 @@
+import { interpolate } from "d3-interpolate";
+
+export const interpolateZerosValue = () => {
+  return Number.MIN_VALUE;
+};
+
+export const interpolateTemporalLinearValue = ({
+  previousValue,
+  nextValue,
+  previousTime,
+  currentTime,
+  nextTime,
+}: {
+  previousValue: number;
+  nextValue: number;
+  previousTime: number;
+  currentTime: number;
+  nextTime: number;
+}) => {
+  return interpolate(
+    previousValue,
+    nextValue
+  )((currentTime - previousTime) / (nextTime - previousTime));
+};
+
+interface TemporalSeriesBeforeImputationEntry {
+  date: Date;
+  value: number | null;
+}
+
+interface TemporalSeriesAfterImputationEntry {
+  date: Date;
+  value: number;
+}
+
+export const imputeTemporalLinearSeries = ({
+  dataSortedByX,
+}: {
+  dataSortedByX: Array<TemporalSeriesBeforeImputationEntry>;
+}): Array<TemporalSeriesAfterImputationEntry> => {
+  const indexedData: Array<[TemporalSeriesBeforeImputationEntry, number]> =
+    dataSortedByX.map((d, i) => [d, i]);
+  const presentDataIndexes = indexedData
+    .filter((d) => d[0].value !== null)
+    .map((d) => d[1]);
+  const missingDataIndexes = indexedData
+    .filter((d) => d[0].value === null)
+    .map((d) => d[1]);
+
+  for (const missingDataIndex of missingDataIndexes) {
+    const nextIndex = presentDataIndexes.findIndex((d) => d > missingDataIndex);
```

Comment:
Nit: The name `nextIndex` is bit confusing because this refers to the index of the array containing indexes.